index.js

```diff
@@ -106,22 +106,126 @@ function selectionChanger(selection,operator,endoperator){
 }
 
 editor.addKeyMap({
-    // bold
-    'Ctrl-B': function(cm) {
-        cm.replaceSelection(selectionChanger(cm.getSelection(),'**'));
-    },
-    // italic
-    'Ctrl-I': function(cm) {
-        cm.replaceSelection(selectionChanger(cm.getSelection(),'_'));
-    },
-    // code
-    'Ctrl-K': function(cm) {
-        cm.replaceSelection(selectionChanger(cm.getSelection(),'`'));
-    },
-    // keyboard shortcut
-    'Ctrl-L': function(cm) {
-        cm.replaceSelection(selectionChanger(cm.getSelection(),'<kbd>','</kbd>'));
+  // bold
+  'Ctrl-B': function(cm) {
+    var selection = cm.getSelection();
+    cm.replaceSelection('**' + selection + '**');
+    if (!selection) {
+      var cursorPos = cm.getCursor();
+      cm.setCursor(cursorPos.line, cursorPos.ch - 2);
     }
+  },
+  // italic
+  'Ctrl-I': function(cm) {
+    var selection = cm.getSelection();
+    cm.replaceSelection('_' + selection + '_');
+    if (!selection) {
+      var cursorPos = cm.getCursor();
+      cm.setCursor(cursorPos.line, cursorPos.ch - 1);
+    }
+  },
+  // code
+  'Ctrl-K': function(cm) {
+    var selection = cm.getSelection();
+    cm.replaceSelection('`' + selection + '`');
+    if (!selection) {
+      var cursorPos = cm.getCursor();
+      cm.setCursor(cursorPos.line, cursorPos.ch - 1);
+    }
+  },
+  // keyboard shortcut
+  'Ctrl-L': function(cm) {
+    cm.replaceSelection(selectionChanger(cm.getSelection(), '<kbd>', '</kbd>'));
```

Comment:
Your changes appear to change the code to be indented with 2 characters instead of 4, which is the current convention of the file. I'd appreciate if you could make sure the file remains consistent.